data/migrations/20190826140654_20190817104145_create-tables.js.js

```diff
@@ -0,0 +1,50 @@
+
+exports.up = function(knex) {
+  return knex.schema
+    .createTable('projects', tbl => {
+        tbl.increments();
+        tbl.text('name', 128)
+            .unique()
+            .notNullable();
+        tbl.text('description', 128);
+        tbl.boolean('completed')
+            .defaultTo(false);
+    })
+    .createTable('resources', tbl => {
+        tbl.increments();
+        tbl.text('name', 128)
+            .unique()
+            .notNullable();
+        tbl.text('description', 128);
+    })
+    .createTable('tasks', tbl => {
+        tbl.increments();
+        tbl.text('description', 128)
+            .notNullable();
+        tbl.text('notes', 128);
+        tbl.boolean('completed')
+            .defaultTo(false);
+        tbl.integer('project_id')
+            .unsigned()
+            .notNullable()
+            .references('projects.id');
+    })
+    .createTable('projects_resources', tbl => {
+        tbl.integer('project_id')
+            .unsigned()
+            .notNullable()
+            .references('projects.id');
+        tbl.integer('resource_id')
+            .unsigned()
+            .notNullable()
+            .references('resources.id');
+    })
+};
+
+exports.down = function(knex) {
+  return knex.schema
+    .dropTableIfExists('projects_resources')
+    .dropTableIfExists('tasks')
+    .dropTableIfExists('resources')
```

Comment:
I would normally switch these two since you drop tables in the reverse order you create them, but I don't think resources references anything in tasks so I think in this instance it's ok